listenTest.py

```diff
@@ -0,0 +1,64 @@
+import time
+from serial_listener import SerialReader  # Adjust this to your actual import path
+
+class SerialDataHandler:
+    '''Class to handle storing received data.'''
+    def __init__(self):
+        # Initialize SerialReader and an empty list to store the data
+        self.reader = SerialReader(timeout=1)
+        self.received_data = []
+
```

Comment:
Code looks good and simple. Do you know where testing fails?

Reply:
I'm guessing the linter errors are due to adding new files outside the normal structure. The plan is to adjust this PR so that it fits with the current project structure using serial port controller.